### PR TITLE
AI Damage Sanity check (Prevents swarmers killing it)

### LINKED ISSUE #7929: Swarmers able to directly kill AI.
**Problem Description**:
Swarmers are able to directly kill the AI with melee stun hits

**What did you expect to happen**: 
Swarmer shouldn't be able to directly kill AI

**What happened instead**:
AI died.

**Why is this bad/What are the consequences:**
Swarmers should not have the ability to directly kill the AI via stun shocks.

**Steps to reproduce the problem**:
Be swarmer. Spam AI with melee stun shocks.

### PR DESCRIPTION
**Pull request by Birdtalon**
Fixes #7929 

Adds Sanity checking to AI receiving damage from `simple_animals` relevant to their damage type.

Prevents swarmers from directly attacking the AI to kill it.

🆑 Birdtalon
fix: Swarmers can no longer kill the AI with direct attacks.
add: Simple animal melee damage now has a sanity check for AI core.
/🆑

### CODE REVIEW

**Comment by Aurorablade** [comment]
this doesn't address the issue of why the type of damage swarmer does damages the AI in the first place.

**Comment by Birdtalon** [comment]
You're correct, it doesn't. The reason why is that the check on the AI doesn't check for damage type at all on melee attack.

It merely randomly takes a value between `melee_damage_lower` `melee_damage_upper` of the `simple_animal` and applies it as Brute damage.

**Comment by tigercat2000** [comment]
This isn't handled particularly well. It'd be better if you resolved the complete ignorance of simple animal damage types in the AI code.

**Comment by Citinited** [comment]
You might also want to look at swarmers being able to ctrl-click the AI to teleport it out of its core and into potentially hazardous areas such as the depowered assembly line.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -697,6 +697,9 @@ var/list/ai_verbs_default = list(
 	return
 
 /mob/living/silicon/ai/attack_animal(mob/living/simple_animal/M)
+	if(istype(M, /mob/living/simple_animal/hostile/swarmer)) //Swarmers shouldn't be killing the AI
+		to_chat(M, "<span class='warning'>Your shock has no effect on [src]!</span>")
+		return
 	if(M.melee_damage_upper == 0)
 		M.custom_emote(1, "[M.friendly] [src]")
 	else
```
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/ai/ai.dm]: Replaced `adjustBruetloess(damage)` with this:

```
		switch(M.melee_damage_type)
			if(BRUTE)
				adjustBruteLoss(damage)
			if(BURN)
				adjustFireLoss(damage)
			if(TOX)
				adjustToxLoss(damage)
			if(OXY)
				adjustOxyLoss(damage)
			if(CLONE)
				adjustCloneLoss(damage)
			if(STAMINA)
				adjustStaminaLoss(damage)
```

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -697,6 +697,9 @@ var/list/ai_verbs_default = list(
 	return
 
 /mob/living/silicon/ai/attack_animal(mob/living/simple_animal/M)
+	if(istype(M, /mob/living/simple_animal/hostile/swarmer)) //Swarmers shouldn't be killing the AI
```
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/ai/ai.dm]: ❄️

Comment (Birdtalon) [inline code/modules/mob/living/silicon/ai/ai.dm]: 🤖 ❄️

**Comment by Birdtalon** [comment]
@Fox-McCloud done.

**Pull request merged**